src/sspi/builders/change_password.rs

```diff
@@ -0,0 +1,109 @@
+use std::cell::RefCell;
+use std::ops::DerefMut;
+
+use crate::{Error, ErrorKind, Result, SecurityBuffer};
+
+pub struct ChangePassword<'a> {
+    pub domain_name: String,
+    pub account_name: String,
+    pub old_password: String,
+    pub new_password: String,
+    pub impersonating: bool,
+    pub output: &'a mut [SecurityBuffer],
+}
+
+#[derive(Default)]
+struct ChangePasswordBuilderInner<'a> {
+    domain_name: Option<String>,
+    account_name: Option<String>,
+    old_password: Option<String>,
+    new_password: Option<String>,
+    impersonating: bool,
+    output: Option<&'a mut [SecurityBuffer]>,
+}
+
+pub struct ChangePasswordBuilder<'a> {
+    inner: RefCell<ChangePasswordBuilderInner<'a>>,
+}
+
+impl<'a> ChangePasswordBuilder<'a> {
+    pub fn new() -> Self {
+        Self {
+            inner: RefCell::new(ChangePasswordBuilderInner::default()),
+        }
+    }
+
+    /// Required
+    pub fn with_domain_name(&self, domain_name: String) -> &Self {
+        self.inner.borrow_mut().domain_name = Some(domain_name);
+        self
+    }
+
+    /// Required
+    pub fn with_account_name(&self, account_name: String) -> &Self {
+        self.inner.borrow_mut().account_name = Some(account_name);
+        self
+    }
+
+    /// Required
+    pub fn with_old_password(&self, old_password: String) -> &Self {
+        self.inner.borrow_mut().old_password = Some(old_password);
+        self
+    }
+
+    /// Required
+    pub fn with_new_password(&self, new_password: String) -> &Self {
+        self.inner.borrow_mut().new_password = Some(new_password);
+        self
+    }
```

Comment:
You could take a `impl Into<String>` instead of a concrete `String`. This is a bit more ergonomic on the call site 🙂